src/compiler/scala/tools/nsc/symtab/classfile/Pickler.scala

```diff
@@ -35,24 +35,50 @@ abstract class Pickler extends SubComponent {
   class PicklePhase(prev: Phase) extends StdPhase(prev) {
     def apply(unit: CompilationUnit) {
       def pickle(tree: Tree) {
-        def add(sym: Symbol, pickle: Pickle) = {
+        def add(sym: Symbol, pickle: Pickle) = withPickle(sym, pickle){ pickle =>
```

Comment:
`withPickle` is called only once -- should we just leave it inlined? The `shouldPickle` condition does occur twice, though.

Reply:
ah, maybe the plan was to factor out the
```
pickle.putSymbol(sym)
currentRun.symData(sym) = pickle
```

part?

Reply:
Oops, `def add` itself was dead code from a failed refactoring attempt. I'll push a cleanup.